airflow/www/views.py

```diff
@@ -1070,6 +1071,47 @@ def clear(self):
 
             return response
 
+    @expose('/dagrun_clear')
+    @login_required
+    @wwwutils.action_logging
+    @wwwutils.notify_owner
+    def dagrun_clear(self):
+        dag_id = request.args.get('dag_id')
+        task_id = request.args.get('task_id')
+        origin = request.args.get('origin')
+        execution_date = request.args.get('execution_date')
+        confirmed = request.args.get('confirmed') == "true"
+
+        dag = dagbag.get_dag(dag_id)
+        execution_date = dateutil.parser.parse(execution_date)
+        start_date = execution_date
+        end_date = execution_date
+
+        if confirmed:
```

Comment:
@aoen  clear here default to include subdags https://github.com/apache/incubator-airflow/blob/master/airflow/models.py#L3149 but adding it definitely makes it more readable :) Will refactor out the logic.